wallet/wallet.go

```diff
@@ -488,7 +488,7 @@ func (sw *SingleAddressWallet) FundV2Transaction(txn *types.V2Transaction, amoun
 	if err := sw.lockUTXOs(toLock); err != nil {
 		return types.ChainIndex{}, nil, fmt.Errorf("failed to lock UTXOs: %w", err)
 	}
-	return sw.tip, toSign, nil
+	return tip, toSign, nil
```

Comment:
Grabbing tip under lock is intentional so that the proof basis in the database doesn't change out from under us.

Reply:
Don't we already have a bit of an issue with `UnspentSiacoinElements` not being called under the lock? We lock the outputs under the lock but we don't fetch them under it. Might also be true for v1 and other places where we call `UnspentSiacoinElements`.